flashlight/nn/modules/PrecisionCast.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <flashlight/autograd/Variable.h>
+#include <flashlight/common/Serialization.h>
+#include <flashlight/common/Utils.h>
+#include <flashlight/nn/modules/Module.h>
+
+namespace fl {
+/**
+ * Precision Cast Module. Casts the input from its original precision to the
+ * target precision. Precision cast only alters the underlying array and leaves
+ * other attributes of the input variable unchanged.
+ */
+class PrecisionCast : public Module {
+  af::dtype targetType_;
+  PrecisionCast() = default;
+  FL_SAVE_LOAD_WITH_BASE(Module, targetType_)
+ public:
+  PrecisionCast(af::dtype targetType);
```

Comment:
I know they're trivial, but can we add doxygen docs for these two?

Would be nice to be precise about the behavior that `std::vector<Variable> forward(const std::vector<Variable>& inputs) override;` casts every input.

Reply:
Sure.